src/RandMsg.chpl

```diff
@@ -589,99 +465,59 @@ module RandMsg
+                                "name: %? size %i dtype: %? state %i".format(name, size, type2str(array_dtype), state));
 
-        st.checkTable(name);
-
-        proc shuffleHelper(type t) throws {
-            var generatorEntry: borrowed GeneratorSymEntry(int) = toGeneratorSymEntry(st.lookup(name), int);
-            ref rng = generatorEntry.generator;
+        var generatorEntry = st[name]: borrowed GeneratorSymEntry(int);
+        ref rng = generatorEntry.generator;
 
-            if state != 1 {
-                // you have to skip to one before where you want to be
-                rng.skipTo(state-1);
-            }
+        if state != 1 then rng.skipTo(state-1);
 
-            ref myArr = toSymEntry(getGenericTypedArrayEntry(xName, st),t).a;
+        const arrEntry = st[xName]: SymEntry(array_dtype, array_nd);
+        ref myArr = arrEntry.a;
+        try {
             rng.shuffle(myArr);
-        }
 
-        select dtype {
-            when DType.Int64 {
-                shuffleHelper(int);
-            }
-            when DType.UInt64 {
-                shuffleHelper(uint);
-            }
-            when DType.Float64 {
-                shuffleHelper(real);
-            }
-            when DType.Bool {
-                shuffleHelper(bool);
-            }
-            otherwise {
-                var errorMsg = "Unhandled data type %s".format(dtypeStr);
-                randLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
-                return new MsgTuple(notImplementedError(pn, errorMsg), MsgType.ERROR);
-            }
         }
-        var repMsg = "created " + st.attrib(xName);
-        randLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),repMsg);
-        return new MsgTuple(repMsg, MsgType.NORMAL);
+        return MsgTuple.success();
     }
 
-    use CommandMap;
-    registerFunction("randomNormal", randomNormalMsg, getModuleName());
-    registerFunction("createGenerator", createGeneratorMsg, getModuleName());
-    registerFunction("uniformGenerator", uniformGeneratorMsg, getModuleName());
-    registerFunction("standardNormalGenerator", standardNormalGeneratorMsg, getModuleName());
-    registerFunction("standardExponential", standardExponentialMsg, getModuleName());
-    registerFunction("segmentedSample", segmentedSampleMsg, getModuleName());
-    registerFunction("choice", choiceMsg, getModuleName());
-    registerFunction("permutation", permutationMsg, getModuleName());
-    registerFunction("poissonGenerator", poissonGeneratorMsg, getModuleName());
-    registerFunction("shuffle", shuffleMsg, getModuleName());
+    proc shuffle(cmd: string, msgArgs: borrowed MessageArgs, st: borrowed SymTab, type array_dtype, param array_nd: int): MsgTuple throws
+        where array_dtype == BigInteger.bigint
+    {
+        return MsgTuple.error("shuffle does not support the bigint dtype");
```

Comment:
I'm not sure how much this matters, but for the methods that operate on the array's incides (like `shuffle` and `permute`) I think any element type is supported since only random integers are needed to generate random indices.